### ISSUE TITLE
Restore smart selection stop on quotes for JS and JSX

### ISSUE THREAD

**Issue by gabrielecirulli** [microsoft/vscode#68072]
Thanks for the great work on version 1.31. After updating, I noticed one of my workflows using the `editor.action.smartSelect.grow` command has had a regression in behavior.

I believe this is related to the recent changes on semantic smart selection. I could find issue https://github.com/Microsoft/vscode/issues/64004 which seems to relate to what I'm reporting.

Some information about my workflow: say I have the following piece of code:

```jsx
const MyComponent = () => (
  <div className="myClassName1 myClassName2">hello</div>
);
```

I would like to place my cursor anywhere inside the `className` attribute's value, then use the `editor.action.smartSelect.grow` command repeatedly until I've selected the entire string, including the quotes: `"myClassName1 myClassName2"`. 

This was possible in version 1.30, as smart selection would first stop at the word (e.g. `myClassName2`) and then expand to the quotes before going past them to the wider context. Version 1.31 doesn't stop at the quotes, and expands past them to the whole element (`<div className...`).

The workflow supported in version 1.30 used to be extremely useful to me as I have a very keyboard-oriented workflow and frequently have to select entire strings, including the quotes, to wrap them in braces or move them somewhere else.

Generally speaking, the aforementioned behavior is also very useful because it facilitates more granular (string) selection while editing, which is impossible in the current version.

I understand the reason this behavior has changed is because smart selection has been delegated to the specific languages. However, the JavaScript language bundled with VSCode does not seem to reimplement this behavior.

Is it possible to restore this functionality for JS/JSX editing?

**Comment by gabrielecirulli** [comment]
> Please also check if it is already covered by an existing one

This issue does not appear to be a duplicate of the ones listed above.

**Comment by jrieken** [comment]
Unfortunate... In the presence of proposed API that allows extensions to properly implement this we have exchanged our default implementation with a simpler one. I have tried to restore the matching-quotes behaviour but that's surprisingly tricky because it depends on how your grammar is defined. Having said that, we will make this proper, proper using the TypeScript syntax tree as the source of truth.
